# Unit 12: **Partition function and its properties. Relation of the partition function to thermodynamic functions.**

# 1. Theory

# 2. Examples

## Problem 1

A molecule can occupy any of the two energy levels with energies of 0 and 600 $cm^{-1}$. At what temperature the number of molecules at the higher energy level is 2 times smaller than the number of molecules at the lower one?

### Solution

In [3]:
import math
import numpy as np

# N0 ~ exp(-E0/kT)
# N1 ~ exp(-E1/kT)
# N1/N0 = exp(-dE/kT) = 0.5, where dE = E1 - E0
# So:
# ln(0.5) = -dE/kT => T = -dE/(k*ln(0.5))

inv_cm2kcal = 0.0028591838441930366  # 4.5563995075520960186520193771085E-6 * 627.5094709  # conversion of cm^{-1} to kcal/mol
kB = 1.9872065E-3 # kcal/mol*K

dE = 600.0 * inv_cm2kcal

T = -dE/(kB * math.log(0.5))

print(T)

1245.4459120399827


### Additional elaboration

#### Step 1: Boltzmann distribution

The ratio of populations is:

$$
\frac{N_1}{N_0} = \exp\left(-\frac{\Delta E}{k_B T}\right)
$$  

We are given:

$$
\frac{N_1}{N_0} = \frac{1}{2}
$$  


#### Step 2: Solve for temperature

Take natural logarithm:

$$
-\frac{\Delta E}{k_B T} = \ln \frac{1}{2} = - \ln 2
$$  

$$
\frac{\Delta E}{k_B T} = \ln 2
$$  

$$
T = \frac{\Delta E}{k_B \ln 2}
$$  


#### Step 3: Convert \($\Delta E\$) to Joules

Given \($\Delta E = 600\ \text{cm}^{-1}$\), convert using:

$$
\Delta E = h c \tilde{\nu}
$$  

where:  
- \($h = 6.626\times10^{-34}\ \text{J·s}$\)  
- \($c = 2.998\times10^{10}\ \text{cm/s}$\)  
- \($\tilde{\nu} = 600\ \text{cm}^{-1}$\)  

$$
\Delta E = 6.626\times10^{-34} \cdot 2.998\times10^{10} \cdot 600 \approx 1.192 \times 10^{-20}\ \text{J}
$$  


#### Step 4: Compute \(T\)

Using \($k_B = 1.381 \times 10^{-23}\ \text{J/K}$\) and \($\ln 2 \approx 0.693$\):

$$
T = \frac{1.192 \times 10^{-20}}{1.381\times10^{-23} \cdot 0.693} \approx 1245\ \text{K}
$$  

## Problem 2

Compute the probabilities of finding atomic sulfur in the ground and in the first excited states (separately) at 1000 K, given the following information about its electronic structure:

| Energy level | Energy, $cm^{-1}$ | Degeneracy |
|-------|------|-------|
| $^3P_2$ |  0  | 5 |
| $^3P_1$ |  396.09  | 3 |
| $^3P_0$ |  573.65  | 1 |
| $^1D_2$ |  9238.58  | 5 |


### Solution

In [4]:
inv_cm2kcal = 0.0028591838441930366 # conversion of cm^{-1} to kcal/mol
kB = 1.9872065E-3 # kcal/mol*K
T = 1000.0 # K

E = np.array([0.0, 396.09, 573.65, 9238.58  ]) * inv_cm2kcal
g = np.array([5.0, 3.0, 1.0, 5.0])

p = g * np.exp(-E/(kB*T))
Z = sum(p)

print(p, Z)

p = p / Z
print(p)

[5.00000000e+00 1.69675867e+00 4.38075040e-01 8.43610572e-06] 7.134842142082729
[7.00786352e-01 2.37813063e-01 6.13994018e-02 1.18238155e-06]


## Problem 3

Compute the probability of finding hydrogen molecule (vibrational frequency is $\omega = 4400\ cm^{-1}$) in a ground vibrational state at T = 4000 K.

### Solution

In [5]:
inv_cm2kcal = 0.0028591838441930366 # conversion of cm^{-1} to kcal/mol
kB = 1.9872065E-3 # kcal/mol*K
T = 4000.0 # K

# Vibrational partition function
hbar_w = 4400.0 * inv_cm2kcal
kBT = kB*T

Z = 1.0/(1.0 - math.exp(-hbar_w/kBT))
print(Z)

# The energy of the grounds state is E = 0, so its contribution to Z is 1, so the probability is:
print(1/Z)

1.2585341711330507
0.7945751676330776


# 3. Problems

# 4. Solutions